pkg/kubectl/describe/versioned/describe.go

```diff
@@ -301,9 +301,16 @@ func printUnstructuredContent(w PrefixWriter, level int, content map[string]inte
 	}
 }
 
-func smartLabelFor(field string) string {
-	commonAcronyms := []string{"API", "URL", "UID", "OSB", "GUID"}
+var commonAcronyms = []string{"API", "URL", "UID", "OSB", "GUID"}
 
+func smartLabelFor(field string) string {
+	// majority of labels and annotations have .io, for example:
+	// controller-tools.k8s.io, failure-domain.beta.kubernetes.io/zone,
+	// deployment.kubernetes.io/revision.
+	// None of these should be camel cased
+	if strings.Contains(field, ".io") {
+		return field
+	}
```

Comment:
Alternatively we could also check for the number of parts, let's say if there are more than 5 than we got something we should not camel case, example is `b805dc46-fc2f-4fb2-9217-3b10401119c3` which due to camelcasing looks like `B 805 Dc 46 - Fc 2 F - 4 Fb 2 - 9217 - 3 B 10401119 C 3` and that's confusing. @smarterclayton @juanvallejo wdyt?

Reply:
I think we could begin by special-casing ".io". I think the two edge cases highlighted in this PR are "strings that look like URLs" and alphanumeric sets separated by dashes ("a8bcd-e12-5ab"). Not sure if we can safely ignore camelCasing based on those two observations though, open to suggestions